### DISCUSSION TITLE
How to delete a lumped port in Excitations of HFSS 3D Modeler?

### DISCUSSION THREAD

**Discussion by passiondong** [ansys/pyaedt#4953] [Q&A] [answered]
I can creat lumped port using hfss.lumped_port, but it seems there's not a delete port function. Could you help me how to delete them? or There is not a delete funciton in pyaedt indeed. Thank you!
![image](https://github.com/user-attachments/assets/81f51cc9-d843-405f-bec9-f5ba3ecb6936)

**Comment by Vamsi-Yaddanapudi** [comment, marked as answer]
Here is the code snipper for deleting the port in HFSS 3D Environment.

import pyaedt
import os

PROJ_NAME = 'Coplaner_Patch_ITO_square.aedt'
PROJ_PATH = r'C:\Users\ykrishna\Downloads\Shared_project'

hfss = pyaedt.Hfss(os.path.join(PROJ_PATH,PROJ_NAME))

boundaries = hfss.boundaries_by_type
lumped_ports = boundaries["Lumped Port"]
#If need to check the name in case of multiple ports
#port_name = lumped_ports[0]._name
lumped_ports[0].delete().

Thanks to Samuel for helping to come up with above script. I have tested and it works

**Comment by passiondong** [comment]
Hi Vamsi,&nbsp;


Thank you very much! I will try it :)


Rgds,
DONG Qin




------------------&nbsp;Original&nbsp;------------------
From: ***@***.***&gt;; 
Date: 2024年8月5日(星期一) 晚上8:08
To: ***@***.***&gt;; 
Cc: ***@***.***&gt;; ***@***.***&gt;; 
Subject: Re: [ansys/pyaedt] How to delete a lumped port in Excitations of HFSS 3D Modeler? (Discussion #4953)














 
Here is the code snipper for deleting the port in HFSS 3D Environment.
 
import pyaedt
 import os
 
PROJ_NAME = 'Coplaner_Patch_ITO_square.aedt'
 PROJ_PATH = r'C:\Users\ykrishna\Downloads\Shared_project'
 
hfss = pyaedt.Hfss(os.path.join(PROJ_PATH,PROJ_NAME))
 
boundaries = hfss.boundaries_by_type
 lumped_ports = boundaries["Lumped Port"]
 #If need to check the name in case of multiple ports
 #port_name = lumped_ports[0]._name
 lumped_ports[0].delete().
 
Thanks to Samuel for helping to come up with above script. I have tested and it works
